src/framework/core/js/FluidRequests.js

```diff
@@ -431,5 +431,236 @@ var fluid_2_0 = fluid_2_0 || {};
+            },
+            "afterRequestComplete.start": {
+                listener: "{that}.start",
+                args: ["{arguments}.1"]
+            }
+        },
+        invokers: {
+            // The add to queue method needs to be provided by the integrator
+            // addToQueue: "",
+            // addToQueue: {funcName: "fluid.identity"},
+            set: {
+                funcName: "fluid.queuedDataSource.enqueueWithModel",
+                args: ["{that}", "{that}.requests.write", "{wrappedDataSource}.set", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+            },
+            get: {
+                funcName: "fluid.queuedDataSource.enqueue",
+                args: ["{that}", "{that}.requests.read", "{wrappedDataSource}.get", "{arguments}.0", "{arguments}.1"]
+            },
+            "delete": {
+                funcName: "fluid.queuedDataSource.enqueue",
+                args: ["{that}", "{that}.requests.write", "{wrappedDataSource}.delete", "{arguments}.0", "{arguments}.1"]
+            },
+            start: {
+                funcName: "fluid.queuedDataSource.start",
+                args: ["{that}", "{arguments}.0"]
+            }
+        }
+    });
+
+    fluid.queuedDataSource.start = function (that, queue) {
+        if (!queue.isActive && queue.requests.length) {
+            var request = queue.requests.shift();
+
+            var requestComplete = function () {
+                queue.isActive = false;
+                that.events.afterRequestComplete.fire(request, queue);
+            };
+
+            queue.isActive = true;
+            var args = "model" in request ? [request.directModel, request.model, request.options] : [request.directModel, request.options];
+            var response = request.method.apply(null, args);
+
+            response.then(requestComplete, requestComplete);
+            fluid.promise.follow(response, request.promise);
+        }
+    };
+
+    /*
+     * Adds only one item to the queue at a time, new requests replace older ones
+     *
+     * The request object contains the request function and arguments.
+     * In the form {method: requestFn, directModel: {}, model: {}, callback: callbackFn}
+     */
+    fluid.queuedDataSource.enqueueImpl = function (that, addToQueueFn, requestsQueue, request) {
+        var promise = fluid.promise();
+        var key = fluid.toHashKey(request.directModel);
+        var queue = requestsQueue[key];
+
+        // add promise to the request object
+        // to be resolved in the start method
+        // when the wrapped dataSource's request returns.
+        request.promise = promise;
+
+        // create a queue if one doesn't already exist
+        if (!queue) {
+            queue = {
+                isActive: false,
+                requests: []
+            };
+            requestsQueue[key] = queue;
+        }
+
+        // queue.requests[0] = request;
+
+        if (typeof(addToQueueFn) === "string") {
```

Comment:
addToQueueFn could never be a string, since it is sourced from an invoker target